org.eclipse.jdt.core.javac/src/org/eclipse/jdt/core/dom/JavacConverter.java

```diff
@@ -142,6 +143,9 @@ void populateCompilationUnit(CompilationUnit res, JCCompilationUnit javacCompila
 		if (javacCompilationUnit.getPackage() != null) {
 			res.setPackage(convert(javacCompilationUnit.getPackage()));
 		}
+		if (javacCompilationUnit.getModule() != null) {
```

Comment:
let's say your project is a java module, i.e there's 1 module-info.java and there are N classes, what does javacCompilationUnit.getModule() mean? Is it:
- only module-info.java has a javacCompilationUnit.getModule() != null, in which case It's probably fine as it is,
- each compilation unit of has a javacCompilationUnit.getModule()? in which case do we really need to create N instance of module, 1 per CU, or can one be shared across all of those CUs?

Reply:
I guess the same question applies for packages

Reply:
I'm pretty sure that only module-info.java has a module.